cli/commands/run-all/command.go

```diff
@@ -17,32 +18,89 @@ import (
 
 const (
 	CommandName = "run-all"
+
+	FlagNameTerragruntProviderCache        = "terragrunt-provider-cache"
+	FlagNameTerragruntProviderCacheDir     = "terragrunt-provider-cache-dir"
+	FlagNameTerragruntProviderCompleteLock = "terragrunt-provider-complete-lock"
+	FlagNameTerragruntRegistryHostname     = "terragrunt-registry-hostname"
+	FlagNameTerragruntRegistryPort         = "terragrunt-registry-port"
+	FlagNameTerragruntRegistryToken        = "terragrunt-registry-token"
+	FlagNameTerragruntRegistryNames        = "terragrunt-registry-names"
 )
 
+func NewFlags(opts *options.TerragruntOptions) cli.Flags {
+	globalFlags := commands.NewGlobalFlags(opts)
+	globalFlags.Add(
+		&cli.BoolFlag{
+			Name:        FlagNameTerragruntProviderCache,
+			Destination: &opts.ProviderCache,
+			EnvVar:      "TERRAGRUNT_PROVIDER_CACHE",
+			Usage:       "Enables provider cache.",
+		},
+		&cli.GenericFlag[string]{
+			Name:        FlagNameTerragruntProviderCacheDir,
+			Destination: &opts.ProviderCacheDir,
+			EnvVar:      "TERRAGRUNT_PROVIDER_CACHE_DIR",
+			Usage:       "The path to the cache directory. Default is .terragrunt-cache/provider-cache in the working directory.",
```

Comment:
Using the same directory by multiple TG processes could cause a conflict, the same as running multiple TF processes with the same PluginCacheDir path. I modified the code, now _Terragrunt Provider Cache_ uses a flock to prevent this. 

Now it uses the native Go function https://pkg.go.dev/os#UserCacheDir + "terragrunt/providers" to determine the default cache directory.